contributors/design-proposals/public-identities-for-pods.md

```diff
@@ -0,0 +1,132 @@
+  instances). Therefore a specialized implementation exposing an IP address per Pod is required.
+
+# API
+
+A new type of Service needs to be created in order to indicate that an external IP address will be exposed per Pod:
+
+```go
+const (
+    //Existing Service types
+	ServiceTypeClusterIP ServiceType = "ClusterIP"
+	ServiceTypeNodePort ServiceType = "NodePort"
+	ServiceTypeLoadBalancer ServiceType = "LoadBalancer"
+	ServiceTypeExternalName ServiceType = "ExternalName"
+
+	//new Service type - expose IP per Pod
+	ServiceTypeIPPerPod ServiceType = "IPPerPod"
+)
+```
+
+A new interface in `cloud.go` will be created to allow creating an external IP addresses.
+
+```go
+// ExternalIP is an abstract, pluggable interface for external ips.
+type ExternalIP interface {
+	// GetExternalIP returns whether the specified external IP exists, and
+	// if so, what its status is.
+	// Implementations must treat the *v1.Service parameter as read-only and not modify it.
+	// Parameter 'clusterName' is the name of the cluster as presented to kube-controller-manager
+	GetExternalIP(clusterName string, service *v1.Service) (status *v1.ExternalIPStatus, exists bool, err error)
+	// EnsureExternalIP creates a new external IP 'name', or updates the existing one.
+	// Returns the status of the external IP.
+	// Implementations must treat the *v1.Service and *v1.Node
+	// parameters as read-only and not modify them.
+	// Parameter 'clusterName' is the name of the cluster as presented to kube-controller-manager
+	EnsureExternalIP(clusterName string, service *v1.Service, nodes []*v1.Node) (*v1.ExternalIPStatus, error)
+	// EnsureExternalIPDeleted deletes the specified external IP if it
+	// exists, returning nil if the external IP specified either didn't exist or
+	// was successfully deleted.
+	// Implementations must treat the *v1.Service parameter as read-only and not modify it.
+	// Parameter 'clusterName' is the name of the cluster as presented to kube-controller-manager
+	EnsureExternalIPDeleted(clusterName string, service *v1.Service) error
+}
+```
+
+Finally a new Annotation will be created to allow falling from "IP per Pod" back into "LoadBalancer per Pod" solution
+(more abot this in the following paragraphs).
+
+```go
+const FallbackToLoadBalancerPerPod = "service.beta.kubernetes.io/fallback-loadbalancer-per-pod"
+```
+
+# Behavior
+
+The logic implemented behind assigning and updating external IPs will be similar to Load Balanced Services.
+
+A ServiceController will periodically iterate though Pods governed by a Service and assign an external IP to each of them.
+Note that there is no `UpdateExternalIP` function in `ExternalIP` interface. This means that a single IP address will not
+be reassigned into another Pod. It will be explicitly deleted (by `EnsureExternalIPDeleted`).
+
+Since `ExternalIP` interface might not be implemented and supported by all Cloud Vendors, we allow creating a Load Balancer
```

Comment:
It doesn't map to any cloud I know.  I get what you're trying to express but it is more complicated than you imagine.  It will required either iptables rules, routing rules, or node ports (or combinations thereof) - which is exactly what Service provides.